Source/Dafny/Dafny.atg

```diff
@@ -2579,46 +2579,78 @@ ExistentialGuard<out Expression e, bool allowLambda>
   Expression<out body, true, allowLambda>
   (. e = new ExistsExpr(x, bvars, null, body, attrs); .)
   .
-MatchStmt<out Statement/*!*/ s>
+
+
+/*------------------------------------------------------------------------*/
+ExtendedPattern<.out ExtendedPattern pat.>
+= (. IToken id; List<ExtendedPattern> arguments;
+    LiteralExpr lit; Expression te; BoundVar bv;
+    pat = null;
+
+  .)
+  ( IF(la.kind == _openparen)
+    "("                                   (. arguments = new List<ExtendedPattern>(); .)
+      [ ExtendedPattern<out pat>          (. arguments.Add(pat); .)
+      { "," ExtendedPattern<out pat>      (. arguments.Add(pat); .)
+      }]
+    ")"                                   (. id = t;
+                                             // make sure the tuple type exists
+                                             theBuiltIns.TupleType(id, arguments.Count, true);
```

Comment:
Report an error here in the parser if the input tried to use a 1-tuple. Dafny does not support 1-tuples, because they are considered to be just parentheses around an expression. A pattern is not allowed to use parentheses, so therefore this should be flagged as an error here.